### ISSUE TITLE
[Button and Link quality] Simplify the `Button` component

### ISSUE THREAD

**Issue by sarahill** [Shopify/polaris-react-archive#4871]
**TL;DR** Update Polaris button and link components with simplified code, purposeful variants, updated styling, and clear usage guidelines.

- [Vault project](https://vault.shopify.io/projects/32613)
- [Project brief](https://docs.google.com/document/d/1HATzkXX1C2H3N3gVsnBOYDIVVpNUx76jlrDZSCcvFHM/edit#heading=h.4iyudv2gmqip)

### What is it and why is it important to Shopify?

Buttons and links are some of the most used primitive Polaris components across admin. These components currently suffer from unnecessary complexity, high number of variations, and lack of composability (see [detailed project brief](https://docs.google.com/document/d/1vc_U-x_DnGRTgUrwHMQnmNZ1odtzME7vT1uJynoRt_w/edit?usp=sharing) for more details). This project will uplift component quality for merchants and improve builder experience for buttons and links by:

- Reducing the number of component variants currently in use
- Simplifying component logic and APIs
- Speeding up the render time of components
- Making components more composable
- Updating visual styles
- Providing clear usage guidance
- Improving the component mobile web experience
- Simplifying the Figma UI Kit components

### What does success look like?

- The number of variants and divergent styles for button and link components is reduced by 50%.
- Component variants and their intended use are clearly documented.
- Differences between button and link usage is clearly documented.
- Button and link wrapping, touch targets, and interactions are optimized for both desktop and mobile web experiences.
- Button component is composable, which will reduce the amount of lines of CSS in custom components.
- Button and link components in the Figma UI Kit use new Figma component features for improved flexibility, ease-of-use, performance, and easier maintenance.
- Bundle size of button and link components is reduced by 30% (research needed).
- Render time of button and link components is 50% faster (research needed).
- Related button and link GitHub issues are closed out (e.g. [Allows a Button to have composable children](https://github.com/Shopify/polaris/issues/6941)).

### Key assumptions

- Shopify wants to invest in mobile web to impact both the web and native experience.
- Investment into button and link components will impact a large percentage of the admin.

### Guiding principles

- All component updates will be aligned with the future direction being defined in the Polaris Uplift project.
- We will not take on system-wide visual changes as part of this project. Visual updates should be isolated to buttons and links and should feel cohesive with our current design language.
- We will migrate Shopify/web and Polaris components to use the new Buttons and Links within this project.
- Starting over with new components will be the better approach for migration and adoption than refactoring the existing components because we will be able to separate adding the new components from the migrations to use them.
- Accessibility is the baseline we will strive for, and we will address known accessibility issues as part of this work.
- We will align the button and link API design between UI Extensions and Polaris as much as possible.

**Comment by sarahill** [comment]
We will be pausing the button work for now and prioritizing a Multiselect Autocomplete component to improve the experience of filtering, adding, and creating tags in the Admin. This pattern is used throughout and there's room to improve it and bring it into Polaris.

Lo wrapped up pulling in and visualizing the component data on the polaris-coverage site and we will use that for future button work and other component work.

Moving this issue to the icebox for now.

**Comment by sarahill** [comment]
This issue is still in progress.

We're currently going through the RFC process for `Link` and working on a draft of the `Button` RFC.

**Comment by sarahill** [comment]
## Summary of the Uplift sync

**TL;DR**
Uplift button designs are ready to go and we can start incorporating the design into the project work. We’ll reach out to Uplift folks as needed for guidance and to keep aligned.

### Goal
Align on what we’ll incorporate into the Button/Link project work from the Uplift work

### Outcome
We determined that some aspects will be out of scope as they are dependent on other aspects of the admin design language.

- Out of scope: `size`, `color`, `border radius`
- In scope: `space`, `depth`, `interactions`
- Transparent vs. Tertiary naming
  - Rename to Tertiary and use level of emphasis to describe how these buttons should be used in the UI.

### Actions items:
- Rename “Transparent” to “Tertiary” in Button RFC + design files
- Corner radius: investigate adding corner smoothing only. This should be time-boxed
- Investigate removing “Large” size buttons
- Explore inverse design and implementation
